starport-blog/04-http-create-post.md

```diff
@@ -0,0 +1,282 @@
+
+## `index.html`
+
+Our `index.html` file will contain only the minimal amount of markup:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>Blog</title>
+  </head>
+  <body>
+    <h1>Blog</h1>
+    <input placeholder="Post title" type="text" id="input" />
+    <button id="button">Create post</button>
+    <div id="posts"></div>
+    <script src="script.js"></script>
+  </body>
+</html>
+```
+
+`<input/>` provides a text input for a title of a new post. `<button>` will submit a new post to our server. `<div id="post>` is an empty container which will be populated by a list of posts. We load `script.js` after all other markup has been loaded.
+
+Now let's create `script.js`.
+
+## `script.js`
+
+Our app will depend on `axios` and `@tendermint/sig`:
+
+```js
+import axios from "axios";
+import {
+  createWalletFromMnemonic,
+  signTx,
+  createBroadcastTx,
+} from "@tendermint/sig";
+```
+
+```js
+const API = "http://localhost:8010/proxy";
+```
+
+HTTP-server runs on `localhost:1317`. Due to CORS we need to route our requests through a proxy: we're using `local-cors-proxy`, which uses the above URL by default.
+
+First, we need to create an account: a public/private key pair, which we will use for signing our (create post) transactions before broadcasting them to the server. We'll be using a mnemonic, from which the keys will be generated.
+
+```js
+const mnemonic =
+  "solid play vibrant paper clinic talent people employ august camp april reduce";
+```
+
+Now let's generate a wallet:
+
+```js
+const wallet = createWalletFromMnemonic(mnemonic);
+```
+
+If you use the same mnemonic (think of it as a password), you will get the same wallet address: in out case `wallet.address` is `cosmos152gzu3vzf7g9tu46vszgpac24lwr48vc8k8kkh`. If you want to generate unique mnemonics for your users, you can use `bip39` package.
+
+Our app will do two things: it will fetch a list of posts and create posts when "Create post" button is clicked.
+
+```js
+const init = () => {
+  fetchPosts();
+  document.getElementById("button").addEventListener("click", createPost);
+};
+```
+
+For `init()` to work we need to define two functions: `fetchPosts` and `createPost`.
+
+```js
+const fetchPosts = () => {
+  axios.get(`${API}/blog/posts`).then(({ data }) => {
+    const posts = JSON.stringify(data.result);
+    document.getElementById("posts").innerText = posts;
+  });
+};
+```
+
+`fetchPosts` makes an HTTP GET request to `/blog/posts` and inserts the posts into the `<div id="posts">` container.
+
+Next, we need to define `createPost` function. It will take the value from the text input, fetch required account parameters from the server, fetch an unsigned transaction from the server, sign it using our private key from the wallet, broadcast it back to the server and fetch the list of posts again.
```

Comment:
```suggestion
Next, we need to define the `createPost` function. It will take the value from the text input, fetch the required account parameters from the server, fetch an unsigned transaction from the server, sign it using our private key from the wallet, broadcast it back to the server and fetch the list of posts again.
```